utils/adapters/firebase.ts

```diff
@@ -0,0 +1,65 @@
+import { App as FirebaseApp } from "firebase-admin/app";
+import { getAuth } from "firebase-admin/auth";
+import { getFirestore } from "firebase-admin/firestore";
+import { randomBytes, secretbox } from "tweetnacl";
+import SolanaAuth, { Adapter } from "../SolanaAuth";
+
+/** FROM: https://github.com/dchest/tweetnacl-js/wiki/Examples */
+const newNonce = () => randomBytes(secretbox.nonceLength);
+
+/**
+ * A SolanaAuth Adapter to connect with Firebase using Firestore
+ * @param {import("firebase-admin/app").App} firebase - A firebase app
+ * @returns {Adapter} Adapter
+ */
+export const FirebaseAdapter = (firebase: FirebaseApp): Adapter => {
+  const auth = getAuth(firebase);
+  const db = getFirestore(firebase);
+
+  return {
+    checkNonce: async (pubkey: string) => {
```

Comment:
I'd rename this to something like

getNonceForPubkey

Or at least getNonce

Because it's no longer checking anything